### PR TITLE
Add ShieldNote.create factory for shield origination

### PR DESCRIPTION
**Pull request by mattgle**
Adds a static `ShieldNote.create` factory so integrators can construct a shield note for a recipient from their already-parsed master public key, instead of only reconstructing notes from scanned on-chain data.

The factory takes `masterPublicKey`, a full pre-fee `value`, ERC20 or ERC721 `tokenData`, and an optional 16-byte `random` (generated via a browser-safe source when omitted). It derives the note public key through the existing `Note.computeNotePublicKey` path and validates token data with the shared `assertValidNoteToken`, so origination-side and scan-side commitment math and validation share one implementation. Inputs are validated up front: key and random lengths, supported token type, all-zero `tokenSubID` for ERC20, value of exactly 1 for ERC721, and a positive value within the contract's uint120 preimage bound. No fee math — the shield fee is deducted on-chain.

Tests include fixed vectors derived with `@railgun-community/circomlibjs` poseidon (and `ethereum-cryptography` keccak256 for the NFT token hash) asserting npk and both ERC20 and ERC721 commitment hashes match the reference implementation, msgpack serialize/deserialize round-trips, and per-rule validation rejections. `deserialize` now normalizes an absent `blockNumber` to `undefined` (msgpack decodes missing fields as `null`), matching the existing `shieldFee` handling.

Stacked on #28; retargets to `dev` once that merges.

### CODE REVIEW

File: src/notes/shield-note.ts [hunk 1]
```diff
@@ -54,6 +68,48 @@ class ShieldNote extends Note {
...
+
+    if (masterPublicKey.length !== 32) {
+      throw new Error(`Master public key must be 32 bytes. Got ${masterPublicKey.length} bytes.`)
+    }
+    if (params.random !== undefined && params.random.length !== 16) {
+      throw new Error(`Random must be 16 bytes. Got ${params.random.length} bytes.`)
+    }
+    if (tokenData.tokenType !== TokenType.ERC20 && tokenData.tokenType !== TokenType.ERC721) {
+      throw new Error(`Unsupported token type for shield note creation: ${tokenData.tokenType}`)
+    }
```
Comment (bhflm) [inline src/notes/shield-note.ts:92]: I think this returns true for empty arrays too, right?

So `ShieldNote.create` could accept `tokenSubID: new Uint8Array(0)` and then we'd end up building a malformed shield request, no?

Can we make this require exactly 32 zero bytes and add tests for empty/short `tokenSubID`?

**Review by bhflm — CHANGES_REQUESTED** [review]
Looks good overall, but I think this one needs a small fix before merge.

in `src/notes/index.ts` if we're exporting `ShieldNoteParams` here, should we also export `ShieldNoteCreateParams`?otherwise the new factory params type isn't available from the barrel export, right?

File: src/notes/shield-note.ts (hunk 1 above)
Comment (mattgle) [reply src/notes/shield-note.ts:92]: ur right

**Review by bhflm — APPROVED** [review]
